### PR TITLE
Public API cleanup for certain IOperation APIs

### PR DESCRIPTION
**Pull request by mavasani**
part of changes listed at https://github.com/dotnet/roslyn/issues/22719

1. Remove the unimplemented IOperation public API GetRootOperation: 
https://github.com/mavasani/roslyn/commit/e3c2994fb4b67018a56561c0db79fd3de9e379b1
2. Remove unused None fields in few public IOperation enums
: https://github.com/mavasani/roslyn/commit/a821d6512f9dce8ce698a038b54ac397df8497db, https://github.com/mavasani/roslyn/commit/303969ebe7d861073691143ec3a13e90d372e5f5 and https://github.com/mavasani/roslyn/commit/3d900354cdf7cd289070a43869c0e112b71af754
3. Add missing xml documentation comments for few public IOperation APIs: https://github.com/mavasani/roslyn/commit/e44145c7141f792d65e283b7d1c9e896dcbc4003

### CODE REVIEW

**Review by AlekseyTs — APPROVED** [review]
LGTM

File: src/Compilers/Core/Portable/Operations/OperationKind.cs [hunk 1]
```diff
@@ -9,6 +9,9 @@ namespace Microsoft.CodeAnalysis
     /// </summary>
     public enum OperationKind
     {
+        /// <summary>
+        /// Indicates an <see cref="IOperation"/> for an not yet implemented semantic construct.
```
Comment (cston) [inline src/Compilers/Core/Portable/Operations/OperationKind.cs]: `for a not yet ...` or perhaps `for a construct that is not implemented yet`.

File: src/Compilers/Core/Portable/Operations/ArgumentKind.cs [hunk 2]
```diff
@@ -7,8 +7,6 @@ namespace Microsoft.CodeAnalysis.Semantics
     /// </summary>
     public enum ArgumentKind
     {
-        None = 0x0,
-
```
Comment (cston) [inline src/Compilers/Core/Portable/Operations/ArgumentKind.cs]: `Explicit` should probably be the default value.

**Review by cston — APPROVED** [review]

**Review by tmat — APPROVED** [review]
:shipit:

File: src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs [hunk 3]
```diff
@@ -2735,15 +2719,15 @@ internal abstract partial class BaseInvocationExpression : Operation, IHasArgume
         /// True if the invocation uses a virtual mechanism, and false otherwise.
         /// </summary>
         public bool IsVirtual { get; }
-        protected abstract ImmutableArray<IArgument> ArgumentsInEvaluationOrderImpl { get; }
+        protected abstract ImmutableArray<IArgument> ArgumentsImpl { get; }
         public override IEnumerable<IOperation> Children
         {
             get
             {
                 yield return Instance;
```
Comment (cston) [inline src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs]: Should `Instance` be skipped if `null`?

File: src/Compilers/Core/Portable/Operations/BranchKind.cs [hunk 4]
```diff
@@ -1,14 +1,30 @@
...
+    /// <summary>
+    /// Kind of the branch for an <see cref="IBranchStatement"/>
+    /// </summary>
     public enum BranchKind
     {
-        None = 0x0,
+        /// <summary>
+        /// Represents unknown branch kind.
+        /// </summary>
+        None = 0x1,
```
Comment (cston) [inline src/Compilers/Core/Portable/Operations/BranchKind.cs]: `= 0x0`

**Review by cston — APPROVED** [review]
A couple of comments. Otherwise, LGTM.

File: src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs (hunk 3 above)
Comment (mavasani) [reply src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs]: @heejaechang is addressing this for all the operations in a separate PR

**Review by jinujoseph — APPROVED** [review]

**Comment by mavasani** [comment]
@jinujoseph - is this pending shiproom approval?

**Comment by jinujoseph** [comment]
yes , this has not been approved yet

**Comment by natidea** [comment]
Approved to merge via [510039](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/510039)

**Pull request merged**